drivers/serial/uart_stm32.c

```diff
@@ -1499,10 +1514,15 @@ static void uart_stm32_irq_config_func_##index(const struct device *dev)	\
 		0);							\
 	irq_enable(DT_INST_IRQN(index));				\
 }
-#else
-#define STM32_UART_IRQ_HANDLER_DECL(index)
+
+#if defined(CONFIG_UART_INTERRUPT_DRIVEN) || defined(CONFIG_UART_ASYNC_API)
+#define STM32_UART_IRQ_HANDLER_FUNC(index)				\
+	.irq_config_func = uart_stm32_irq_config_func_##index,
+#define STM32_UART_POLL_IRQ_HANDLER_FUNC(index)
+#elif defined(CONFIG_PM)
 #define STM32_UART_IRQ_HANDLER_FUNC(index)
-#define STM32_UART_IRQ_HANDLER(index)
+#define STM32_UART_POLL_IRQ_HANDLER_FUNC(index)				\
```

Comment:
in case the soc (like stm32l476) has no CONFIG_UART_INTERRUPT_DRIVEN, no CONFIG_UART_ASYNC_API, no CONFIG_PM, then the build is raising an error:
`drivers/serial/uart_stm32.c:1547:3: error: implicit declaration of function 'STM32_UART_IRQ_HANDLER_FUNC'; did you mean 'STM32_UART_IRQ_HANDLER_DECL'? [-Werror=implicit-function-declaration]`
--> for example:  _west build -p auto -b nucleo_l476rg `tests/kernel/timer/timer_api/_

Reply:
good catch, should be fixed